api/exceptions.inc

```diff
@@ -1,8 +1,14 @@
 <?php
 
 // Base exception that all API exceptions should derive from
+// These exception codes range from 1 to 99
 class ApiException extends Exception
 {
+    public function __construct($message = "API exception", $code = 1)
```

Comment:
For this and all similar instances -- are these messages supposed to be translatable?

Reply:
As I understand it, we should translate messages which are shown to the end user when something 'expected' happens, like when there are no more pages left to proofread. Some messages would only apear if something unexpected went wrong in the server or the client application. Then the message would be more use to someone looking for the bug who might be more likely to understand it if left in English.
There is one case which is affected by this PR: if you select a different project in the page browser and put in an invalid or nonexistent code then formerly you would get `No such project ID` not translatable, but now get `There is no project with projectid 'asdf'` which is translatable.
Composing this reply I tested whether I can trigger another error by manipulating the url to get a nonexistent page (and press return) when the error shown will be `There is no page 00.png in this project` translated. This message comes from the client rather than the server. (I'm not sure that the client should really work like this but this is a different issue.)

I'll make a sandbox should you wish to try it out.

Reply:
I would have thought that `No such projectID` would be translatable? Or do you mean that it's just in a place where it can't be translated.

Thanks for the sandbox! I tried both your tests, and they look good.

I know you've mentioned translation before -- how will that work with the messages coming from the client instead of the server. And of course ... the answer is not critical at this point, so I'm happy for you to say "we'll get to that later!". 😁

Reply:
`No such projectID` could have been translated but wasn't.

At present the client messages are downloaded from the server beforehand, having been translated. So far we have just implemented a simple substitution of one %s in some cases like the one above.